tests/utils/node_launcher.go

```diff
@@ -0,0 +1,75 @@
+package utils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/kurtosis-tech/kurtosis/api/golang/engine/lib/kurtosis_context"
+)
+
+const (
+	isPartitioningEnabled    = false
+	enclaveIdPrefix          = "test"
+	avalancheStarlarkPackage = "github.com/kurtosis-tech/avalanche-package"
+	// forces the node to launch on 9650 instead of ephemeral ports
+	forceExposeOn9650  = `{"test_mode": true}`
+	defaultParallelism = 4
+	firstNodeId        = "node-0"
+)
+
+func SpinupAvalancheNode() (string, func(), error) {
+	ctx := context.Background()
+
+	kurtosisCtx, err := kurtosis_context.NewKurtosisContextFromLocalEngine()
+	if err != nil {
+		return "", nil, err
+	}
+
+	enclaveId := fmt.Sprintf("%s-%d", enclaveIdPrefix, time.Now().Unix())
+	enclaveCtx, err := kurtosisCtx.CreateEnclave(ctx, enclaveId, isPartitioningEnabled)
+	if err != nil {
+		return "", nil, err
+	}
+
+	runResult, err := enclaveCtx.RunStarlarkRemotePackageBlocking(ctx, avalancheStarlarkPackage, forceExposeOn9650, false, defaultParallelism)
+	if err != nil {
+		return "", nil, err
+	}
+
+	if runResult.InterpretationError != nil {
+		return "", nil, errors.New("error interpreting Starlark code")
+	}
+	if len(runResult.ValidationErrors) != 0 {
+		return "", nil, errors.New("error validating Starlark code")
+	}
+	if runResult.ExecutionError != nil {
+		return "", nil, errors.New("error executing Starlark code")
+	}
```

Comment:
I have propagated the errors. We check three different types of errors as Kurtosis executes in [Multiple Phases](https://docs.kurtosis.com/explanations/why-multi-phase-runs)

Perhaps our SDK could have an API that abstracts away the errors and returns `scriptOutput, err` where err is true regardless of it being interpretation, validation, execution or any kind of errors. So far we have been following a pattern where we assert all of them are nils individually like [here](https://github.com/kurtosis-tech/awesome-kurtosis/blob/main/quickstart/go-test/main_test.go#L64-L67)

Reply:
We spent some cycles this morning and have changed the API so that only one error needs to be checked and it propagates the right error to you. Making it simpler for people to use the Go SDK.